asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py

```diff
@@ -0,0 +1,359 @@
+    )
+
+    posit_confidence_cutoff: PositiveFloat = Field(
+        0.7, description="POSIT confidence cutoff used to filter docking results"
+    )
+
+    logname: str = Field("large_scale_docking.log", description="Name of the log file.")
+
+    loglevel: int = Field(logging.INFO, description="Logging level.")
+
+    output_dir: Path = Field(Path("output"), description="Output directory")
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @validator("posit_confidence_cutoff")
+    @classmethod
+    def posit_confidence_cutoff_must_be_between_0_and_1(cls, v):
+        """
+        Validate that the POSIT confidence cutoff is between 0 and 1
+        """
+        if v < 0 or v > 1:
+            raise ValueError("POSIT confidence cutoff must be between 0 and 1.")
+        return v
+
+    @root_validator
+    @classmethod
+    def check_inputs(cls, values):
+        """
+        Validate inputs
+        """
+        filename = values.get("filename")
+        fragalysis_dir = values.get("fragalysis_dir")
+        structure_dir = values.get("structure_dir")
+        postera = values.get("postera")
+        postera_upload = values.get("postera_upload")
+        postera_molset_name = values.get("postera_molset_name")
+        du_cache = values.get("du_cache")
+        gen_du_cache = values.get("gen_du_cache")
+
+        if postera and filename:
+            raise ValueError("Cannot specify both filename and postera.")
+
+        if not postera and not filename:
+            raise ValueError("Must specify either filename or postera.")
+
+        if postera_upload and not postera_molset_name:
+            raise ValueError(
+                "Must specify postera_molset_name if uploading to postera."
+            )
+
+        if fragalysis_dir and structure_dir:
+            raise ValueError("Cannot specify both fragalysis_dir and structure_dir.")
+
+        if not fragalysis_dir and not structure_dir:
+            raise ValueError("Must specify either fragalysis_dir or structure_dir.")
+
+        if du_cache and gen_du_cache:
+            raise ValueError("Cannot specify both du_cache and gen_du_cache.")
+
+        return values
+
+    @validator("du_cache")
+    @classmethod
+    def du_cache_must_be_directory(cls, v):
+        """
+        Validate that the DU cache is a directory
+        """
+        if v is not None:
+            if not Path(v).is_dir():
+                raise ValueError("Du cache must be a directory.")
+        return v
+
+
+def large_scale_docking(inputs: LargeScaleDockingInputs):
+    """
+    Run large scale docking on a set of ligands, against multiple targets
+    """
+
+    output_dir = inputs.output_dir
+    if output_dir.exists():
+        rmtree(output_dir)
+    output_dir.mkdir()
+
+    logger = FileLogger(
+        inputs.logname, path=output_dir, stdout=True, level=inputs.loglevel
+    ).getLogger()
+
+    logger.info(f"Running large scale docking with inputs: {inputs}")
```

Comment:
We could also dump the JSON of the inputs to the output folder that will make it easy to run again by just loading the inputs from that file.

Reply:
Yep good idea.